paymentsheet/src/main/java/com/stripe/android/paymentsheet/forms/FormViewModel.kt

```diff
@@ -28,27 +33,55 @@ import javax.inject.Singleton
  * @param: layout - this contains the visual layout of the fields on the screen used by [Form]
  * to display the UI fields on screen.  It also informs us of the backing fields to be created.
  */
-@Singleton
 internal class FormViewModel @Inject internal constructor(
     layout: LayoutSpec,
     config: FormFragmentArguments,
     private val resourceRepository: ResourceRepository
 ) : ViewModel() {
     internal class Factory(
-        private val resources: Resources,
-        private val layout: LayoutSpec,
-        private val formArguments: FormFragmentArguments
-    ) : ViewModelProvider.Factory {
+        val config: FormFragmentArguments,
+        val resource: Resources,
+        var layout: LayoutSpec
+    ) : ViewModelProvider.Factory, Injectable<Factory.FallbackInitializeParam> {
+        internal data class FallbackInitializeParam(
+            val resource: Resources
+        )
+
+        @Inject
+        lateinit var subComponentBuilderProvider: Provider<FormViewModelSubcomponent.Builder>
+
         @Suppress("UNCHECKED_CAST")
         override fun <T : ViewModel?> create(modelClass: Class<T>): T {
 
-            // This is where we will call Dagger:
-            return DaggerFormViewModelComponent.builder()
-                .resources(resources)
+            val logger = Logger.getInstance(BuildConfig.DEBUG)
+            WeakMapInjectorRegistry.retrieve(config.injectorKey)?.let {
+                logger.info(
+                    "Injector available, " +
+                        "injecting dependencies into FormViewModel.Factory"
+                )
+                it.inject(this)
+            } ?: run {
+                logger.info(
+                    "Injector unavailable, " +
+                        "initializing dependencies of FormViewModel.Factory"
+                )
+                fallbackInitialize(
```

Comment:
Did we want to add a test for this branching or is that something hard to do?

Reply:
+1, please refer to the test [here](https://github.com/stripe/stripe-android/blob/2989d01d432329015b1eb31628c996cfd4c66735/paymentsheet/src/test/java/com/stripe/android/paymentsheet/PaymentOptionsViewModelTest.kt#L230) to verify the factory could create a ViewModel successfully with/without Injector